ui/DesignSystem/Component/Wallet/Transactions/TxListItem.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="typescript">
+  import type { Tx } from "ui/src/transaction";
+  import { Icon } from "ui/DesignSystem/Primitive";
+  // import { Label } from "ui/DesignSystem/Component";
+  import { getDay, getShortMonth, txIcon } from "ui/src/transaction";
+
+  export let tx: Tx;
+
+  // const price = 0.0043;
+</script>
+
+<style>
+  .container {
+    display: flex;
+    padding: 0.75rem;
+    border-bottom: 1px solid var(--color-foreground-level-2);
+    align-items: center;
+    justify-content: space-between;
+    cursor: pointer;
+  }
+
+  .container:last-child {
+    border-bottom: 0;
+  }
+
+  .left {
+    display: flex;
+    align-items: center;
+  }
+
+  .date {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    width: 2rem;
+    margin-right: 1rem;
+  }
+
+  .date h5 {
+    color: var(--color-foreground-level-4);
+    line-height: 1rem;
+  }
+
+  .date p {
+    color: var(--color-foreground-level-5);
+  }
+</style>
+
+<div class="container" on:click>
+  <div class="left">
+    <div class="date">
+      <h5>{getShortMonth(new Date(tx.date))}</h5>
+      <p class="typo-text-bold">{getDay(new Date(tx.date))}</p>
+    </div>
+    <svelte:component this={Icon[txIcon(tx)]} />
```

Comment:
@geigerzaehler @rudolfs how do I typescript this `{Icon[txIcon(tx)]}` thing?

Reply:
You can do this by having `txIcon` return the actual icon instead of a string identifying the icon.

```ts
import type { SvelteComponent } from "svelte";

export function txIcon(tx: Tx): typeof SvelteComponent {
  switch (tx.kind) {
    case TxKind.ClaimRadicleIdentity:
      return Icon.Registered;
    // ...
  }
}
```

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/1900/commits/533e08baeb0e6e03c45f213979c8f96ab9bd9ccd